api/v1alpha1/gateway_types.go

```diff
@@ -54,78 +55,212 @@ type GatewayList struct {
 type GatewaySpec struct {
 	// Class used for this Gateway. This is the name of a GatewayClass resource.
 	Class string `json:"class" protobuf:"bytes,1,opt,name=class"`
-	// Listeners associated with this Gateway. Listeners define what addresses,
-	// ports, protocols are bound on this Gateway.
+
+	// Listeners associated with this Gateway. Listeners define
+	// logical endpoints that are bound on this Gateway's addresses.
+	// At least one Listener MUST be specified.
+	//
+	// Each Listener in this array must have a unique Port field,
+	// however a GatewayClass may collapse compatible Listener
+	// definitions into single implementation-defined acceptor
+	// configuration even if their Port fields would otherwise conflict.
+	//
+	// Listeners are compatible if all of the following conditions are true:
```

Comment:
We can fix this in a later commit, but it would be good to frame this from the user's standpoint, as the current description is somewhat hard to parse.